### PR TITLE
[16.0][IMP] base_wamas_ubl: Add record_to_wamas method

### PR DESCRIPTION
**Pull request by QuocDuong1306**
After reviewing, i will :
- Close `partner_export_wamas`, `product_export_wamas`
- Update `allo_edi_partner_export_wamas` and `allo_edi_product_export_wamas`

### CODE REVIEW

File: base_wamas_ubl/models/base_wamas_ubl.py [hunk 1]
```diff
@@ -42,3 +42,17 @@ def get_wamas_type(self, str_file):
     @api.model
     def wamas2wamas(self, str_file):
         return wamas2wamas(str_file)
+
+    @api.model
+    def record_to_wamas(self, specific_dict, telegram=False):
```
Comment (nilshamerlinck) [inline base_wamas_ubl/models/base_wamas_ubl.py]:
```suggestion
    def record_to_wamas(self, specific_dict, telegram_type):
```

File: base_wamas_ubl/models/base_wamas_ubl.py [hunk 2]
```diff
@@ -42,3 +42,17 @@ def get_wamas_type(self, str_file):
...
+        """
+        Export data as WAMAS format
+
+        :return: data as WAMAS format
+        :rtype: bytes
+        """
+        if not telegram:
+            raise ValueError(_("Need telegram for exporting!"))
```
Comment (nilshamerlinck) [inline base_wamas_ubl/models/base_wamas_ubl.py]:
```suggestion
            raise ValueError(_("Please define telegram_type."))
```

File: base_wamas_ubl/models/base_wamas_ubl.py [hunk 3]
```diff
@@ -42,3 +42,17 @@ def get_wamas_type(self, str_file):
...
+
+    @api.model
+    def record_to_wamas(self, specific_dict, telegram=False):
+        """
+        Export data as WAMAS format
+
+        :return: data as WAMAS format
+        :rtype: bytes
+        """
+        if not telegram:
```
Comment (nilshamerlinck) [inline base_wamas_ubl/models/base_wamas_ubl.py]:
```suggestion
        if not telegram_type:
```

File: base_wamas_ubl/models/base_wamas_ubl.py [hunk 4]
```diff
@@ -42,3 +42,17 @@ def get_wamas_type(self, str_file):
...
+            raise ValueError(_("Need telegram for exporting!"))
+        if not isinstance(specific_dict, dict):
+            raise ValueError(_("The specific dict is not valid."))
+        return self.dict2wamas(specific_dict, telegram)
```
Comment (nilshamerlinck) [inline base_wamas_ubl/models/base_wamas_ubl.py]:
```suggestion
        return self.dict2wamas(specific_dict, telegram_type)
```

**Review by nilshamerlinck — CHANGES_REQUESTED** [review]

**Comment by QuocDuong1306** [comment]
Thank @nilshamerlinck , I updated

**Review by nilshamerlinck — APPROVED** [review]

**Pull request closed without merging**